packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts

```diff
@@ -183,6 +189,198 @@ export const getAllLatestEntitiesResolver: ResolverFn<
   return entitySubgraph as Subgraph;
 };
 
+export const aggregateEntitiesResolver: Extract<
```

Comment:
Unfortunately we need this `Extract`. The generated type is a `Resolver<…>` which is:
```ts
export type Resolver<TResult, TParent = {}, TContext = {}, TArgs = {}> =
  | ResolverFn<TResult, TParent, TContext, TArgs>
  | ResolverWithResolve<TResult, TParent, TContext, TArgs>;
```

Where `ResolverFn` is a function (duh) and `ResolverWithResolve` is 
```ts
export type ResolverWithResolve<TResult, TParent, TContext, TArgs> = {
  resolve: ResolverFn<TResult, TParent, TContext, TArgs>;
};
```

So if I use the generated type, this then causes problems in our middleware-types
```ts
export type ResolverMiddleware<
  TStartContext,
  TArgs,
  TEndContext = TStartContext,
> = (
  next: ResolverFn<any, any, TEndContext, any>,
) => ResolverFn<any, any, TStartContext, TArgs>;
```

Changing that to take `Resolver` instead would be quite a pain to deal with as we’d have to type-guard to check if it’s an object everywhere we use it

Reply:
This still applies in the rebased PR